spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/ExpandDims.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.nn.Unsqueeze
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.tensor._
+
+import scala.reflect.ClassTag
+
+class ExpandDims[T: ClassTag](axis: Int)
+  (implicit ev: TensorNumeric[T]) extends Operation[Tensor[_], Tensor[_], T] {
+  var squeeze: Unsqueeze[_] = _
+
+  override def updateOutput(input: Tensor[_]): Tensor[_] = {
+    if (squeeze == null || squeeze.getNumericType() != input.getType()) {
+      squeeze = input.getType() match {
+        case FloatType =>
+          Unsqueeze[Float](axis + 1)
```

Comment:
It should be alway Unsqueeze[T]. I think we should modify Unsqueeze to make it handle Tensor[_]

Reply:
If we modify Unsqueeze to make it handle Tensor[_], then the T would not be useful to contain the input and output tensor type.